utility/data_fetcher.go

```diff
@@ -145,6 +145,9 @@ func (d *ManualDecoder) MapStr(base map[string]interface{}, key string, keys ...
 
 func (d *ManualDecoder) TimeRFC3339(base map[string]interface{}, key string, keys ...string) time.Time {
 	val := getDeep(base, keys...)[key]
+	if val == nil {
+		return time.Time{}
```

Comment:
How about introducing a `TimeRFC3339WithDefault` method that returns a `time.Now()` instead of a zero time in case it is empty? The decoding happens before the transformation, so the timestamp would be more accurate, and no further handling in the `transform` function would be needed.